devtools/gradle/gradle-application-plugin/src/main/java/io/quarkus/gradle/tasks/QuarkusGradleUtils.java

```diff
@@ -46,44 +46,35 @@ public static String getClassesDir(SourceSet sourceSet, File tmpDir, boolean pop
     }
 
     public static Path mergeClassesDirs(Collection<Path> classesDirs, File tmpDir, boolean populated, boolean test) {
-        Path classesDir = null;
-        final Iterator<Path> i = classesDirs.iterator();
-        int dirCount = 0;
-        while (i.hasNext()) {
-            final Path next = i.next();
-            if (!Files.exists(next)) {
-                continue;
+        List<Path> existingClassesDirs = classesDirs.stream().filter(p -> Files.exists(p)).toList();
+
+        if (existingClassesDirs.size() == 0) {
+            return null;
+        }
+
+        if (existingClassesDirs.size() == 1) {
+            return existingClassesDirs.iterator().next();
```

Comment:
```suggestion
            return existingClassesDirs.get(0);
```

Reply:
Ah yes, I made it a list in the end, it was initially a collection, will fix!